waspc/cli/Command/Telemetry/Project.hs

```diff
@@ -89,17 +92,31 @@ initialCache = ProjectTelemetryCache { _lastCheckIn = Nothing, _lastCheckInBuild
 
 -- * Project telemetry cache file.
 
-readOrCreateProjectTelemetryFile :: Path Abs (Dir TelemetryCacheDir) -> ProjectHash -> IO ProjectTelemetryCache
-readOrCreateProjectTelemetryFile telemetryCacheDirPath projectHash = do
-    fileExists <- SD.doesFileExist filePathFP
-    maybeCache <- if fileExists then readCacheFile else return Nothing
-    case maybeCache of
-        Just cache -> return cache
-        Nothing -> writeProjectTelemetryFile telemetryCacheDirPath projectHash initialCache >> return initialCache
+convertMaybeTimeToString :: Maybe T.UTCTime -> String
+convertMaybeTimeToString = maybe "nil" show
+
+showLastSentTelemetryDateTime :: Maybe ProjectTelemetryCache -> String
```

Comment:
This output (strings) feels very specific to that CLI command in Command/Telemetry.hs, so I wouldn't put it here.
Instead, I would implement and export following function here:
```hs
getTimeOfLastTelemetryDataSent :: ProjectTelemetryCache -> T.UTCTime
```
which would return the latest time of two times: _lastCheckIn and _lastCheckInBuild. 

We would then use this in Command/Telemetry.hs to inform user about the last time telemetry data was sent (without explaining the difference between build no-build data - we just report last time something was sent and that is it).

Also, if no data was sent yet, don't print `nil` where time would go. Instead, it might be best if we don't print anything, since cache might have been deleted and that means we just don't know when and if the last data was sent.